### ISSUE TITLE
RangeError: "size" argument must not be larger than 2147483647

### ISSUE THREAD

**Issue by ckylape** [Zazama/node-id3#38]
I'm getting the following error when trying to read tags from a 9.7MB MP3:

```
RangeError: "size" argument must not be larger than 2147483647
    at Function.Buffer.alloc (buffer.js:229:3)
    at new Buffer (buffer.js:152:19)
    at NodeID3.getTagsFromBuffer (.../node_modules/node-id3/index.js:318:31)
    at NodeID3.read (.../node_modules/node-id3/index.js:226:25)
    at Object.<anonymous> (.../index.js:40:24)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
```

**Snippet:**

```javascript
const file = 'sample.mp3'

const buf = fs.readFileSync(file)
console.log(buf.length)
// 9673454 

try {
    let tags = NodeID3.read(file)
} catch (err) {
    console.log(err)
}
```

**Environment:**

* OSX - 10.12.6
* Node - 8.5.0
* node-id3 - 0.1.3

**Update:**

I added `console.log(bodyFrameHeader.toString('utf8', 0, 4) + ' ' + bodyFrameSize)` to try to debug what was flagging the size limit; here's the output:

```
TPE2 29
TPOS 9
WCOP 17
WOAR 40
TLEN 4
TMED 29
TXXX 29
TXXX 31
TXXX 39
TXXX 59
TXXX 31
TXXX 37
TXXX 25
GEOB 100
TXXX 31
GEOB 110
PRIV 334456
�� 3363242937
```

**Comment by Zazama** [comment]
Hello,
would you mind sending me the .mp3 file you're having troubles with to the email in the package.json?

**Comment by ckylape** [comment]
@Zazama - I sent the MP3 file. I also did some digging and found the culprit to be the cover art. 

The file I sent has two cover art photos. The first is the true cover art which is 1 MB @ 1400x1400, the second is a wave pattern added by [Beatport Pro](https://www.beatport.com/desktop) which is 51 KB @ 1200x200. 

If I remove either of the cover art, the script works fine. 

Maybe something getting multiplied? The max buffer size should be 2GB.

**Comment by Zazama** [comment]
Yeah, I took a small look at it.
Seems like there are two bugs:
1. the second cover art frame isn't read at its start (maybe it's even trying to read in the middle of the first cover art frame)
2. due to this, it reads random 4 bytes which can hold an extremly high number for the frame size, but the library should check if this size of a frame is bigger than the rest of the ID3-Tag.

I'm a bit sick right now, so I won't dig deeper until I'm better, but a fix shouldn't be too complicated when I'm on it. I could speculate and say it might be because of two present APICs (cover arts) which I've never tested before.

Sorry for the wait, hope it's not urgent.

**Comment by towfiqi** [comment]
@Zazama Hope you are ok now. Is it possible to avoid reading the cover art data to prevent this issue as I am not utilizing cover arts in my app.

**Comment by Zazama** [comment]
Fixed https://github.com/Zazama/node-id3/commit/d16530a2364f2b4b34104ef5b2c08c4bd906b57e and published to npm with 0.1.6

**Comment by towfiqi** [comment]
Great! Thanks for the quick fix :)
